state_synchronizer/src/peer_manager.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::PeerId;
+use std::collections::HashMap;
+use types::account_address::AccountAddress;
+
+#[derive(Default)]
+#[allow(dead_code)]
```

Comment:
why need to do that?

Reply:
otherwise I get field never used error

Reply:
we can add use of it in this diff
so actually set it on by default and during pick routine check that you select from active upstream nodes only

Reply:
You mean in pick_peer in peer_manager.rs we want to first filter by peers that are active and upstream first, then choose a random one out of those?